### PR TITLE
Auth0 Frontend Implementation

### PR DESCRIPTION
**Pull request by NathHorrigan**
- Login with email/pass
- Signup with email/pass
- Forgot pass function that sends email
- Redux state saved so session is preserved.
- Auto-logout once token expires (Not final atm, should renew)

### CODE REVIEW

**Comment by tm-kn** [comment]
Issue - #16

File: mresponse/frontend/app/src/components/avatar/index.js [hunk 1]
```diff
@@ -11,7 +11,7 @@ const Avatar = ({ src, className = '', editable, onClick }) => {
         <div className="avatar-edit" onClick={() => onClick()}>
           <img
             className="avatar-edit-icon"
-            src="staticmedia/icons/camera.svg"
+            src="/static/media/icons/camera.svg"
```
Comment (tm-kn) [inline mresponse/frontend/app/src/components/avatar/index.js]: Can't we have a function to generate static URLs? Currently it's basically hardcoded. We'd rather do `static('media/icons/camera.svg`), I guess.

File: mresponse/frontend/app/src/components/login-form/index.js [hunk 2]
```diff
@@ -6,7 +6,7 @@ import HighlightedText from '@components/highlighted-text'
 import Button from '@components/buttons'
 
 export default class LoginForm extends React.Component {
-  state = { status: "What's your email address?", email: '', password: '' }
+  state = { status: this.props.status || "What's your email address?", email: '', password: '' }
```
Comment (tm-kn) [inline mresponse/frontend/app/src/components/login-form/index.js]: Does it actually pass `this` in there? I am not familiar with this ES6 syntax very much.  That's executed in the constructor I guess?

File: .gitignore [hunk 3]
```diff
@@ -33,3 +33,4 @@ wheels/
 # Frontend Tooling
 mresponse/frontend/app/.out/
 *.log
+.env
```
Comment (tm-kn) [inline .gitignore]: Can you please set your editor to use unix line endings please?

File: mresponse/frontend/app/src/components/avatar/index.js (hunk 1 above)
Comment (NathHorrigan) [reply mresponse/frontend/app/src/components/avatar/index.js]: Yes, I need to do do this to fix it for storybook as well. Next on my list.

File: mresponse/frontend/app/src/components/login-form/index.js (hunk 2 above)
Comment (NathHorrigan) [reply mresponse/frontend/app/src/components/login-form/index.js]: Yeh, it works fine.

File: mresponse/frontend/urls.py [hunk 4]
```diff
@@ -18,5 +18,5 @@
         ),
         name='manifest.json',
     ),
-    path('', TemplateView.as_view(template_name="index.html"))
+    re_path('', TemplateView.as_view(template_name="index.html"))
```
Comment (tm-kn) [inline mresponse/frontend/urls.py]: I think to make it catch-all you need to do it this way.
```
r'^.*/$
```

I think.

File: mresponse/frontend/app/.env.example [hunk 5]
```diff
@@ -0,0 +1,3 @@
+REACT_APP_AUTH_DOMAIN =
+REACT_APP_AUTH_CLIENT_ID =
+REACT_APP_AUTH_CALLBACK_URL =
```
Comment (tm-kn) [inline mresponse/frontend/app/.env.example]: Another file without line ending.

**Pull request merged**